flake.nix

```diff
@@ -24,372 +23,81 @@
-              (fetchpatch {
-                name = "fix-build-gcc-12.patch";
-                url =
-                  "https://github.com/ghdl/ghdl/commit/f8b87697e8b893b6293ebbfc34670c32bfb49397.patch";
-                hash = "sha256-tVbMm8veFkNPs6WFBHvaic5Jkp1niyg0LfFufa+hT/E=";
-              })
-            ];
-
-            LIBRARY_PATH = "${stdenv.cc.libc}/lib";
-
-            nativeBuildInputs = [ gnat ];
-            buildInputs = [ zlib ]
-              ++ lib.optionals (backend == "llvm") [ llvm ];
-            propagatedBuildInputs = [ ]
-              ++ lib.optionals (backend == "llvm") [ zlib ];
 
-            preConfigure = ''
-              # If llvm 7.0 works, 7.x releases should work too.
-              sed -i 's/check_version  7.0/check_version  7/g' configure
-            '';
-
-            configureFlags = [
-              # See https://github.com/ghdl/ghdl/pull/2058
-              "--disable-werror"
-              "--enable-synth"
-            ] ++ lib.optionals (backend == "llvm")
-              [ "--with-llvm-config=${llvm.dev}/bin/llvm-config" ];
-
-            hardeningDisable = [ "format" ];
-
-            enableParallelBuilding = true;
-
-            passthru = {
-              # run with either of
-              # nix-build -A ghdl-mcode.passthru.tests
-              # nix-build -A ghdl-llvm.passthru.tests
-              tests = {
-                simple = callPackage ./test-simple.nix { inherit backend; };
-              };
-            };
-
-            meta = with lib; {
-              homepage = "https://github.com/ghdl/ghdl";
-              description = "VHDL 2008/93/87 simulator";
-              maintainers = with maintainers; [ lucus16 thoughtpolice ];
-              platforms = platforms.linux;
-              license = licenses.gpl2;
-            };
-          };
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) lib callPackage stdenv fetchgit fetchFromGitHub;
+        in rec {
+          default = yosys;
 
-        abc-verifier = with final;
-          stdenv.mkDerivation rec {
-            pname = "abc-verifier";
-            version = "for-yosys-0.17";
+          ghdl = pkgs.ghdl;
 
+          abc-verifier = pkgs.abc-verifier.overrideAttrs (_: rec {
+            version = "yosys-0.17";
             src = fetchFromGitHub {
               owner = "yosyshq";
               repo = "abc";
-              rev = "09a7e6d";
+              rev = "09a7e6dac739133a927ae7064d319068ab927f90" # == version
+              ;
               hash = "sha256-+1UcYjK2mvhlTHl6lVCcj5q+1D8RUTquHaajSl5NuJg=";
             };
-
-            nativeBuildInputs = [ cmake ];
-            buildInputs = [ readline ];
-
-            installPhase = "mkdir -p $out/bin && mv abc $out/bin";
-
-            # needed by yosys
             passthru.rev = src.rev;
+          });
```

Comment:
Specific version of abc-verifier is required to work with yosys 0.17. Our changes consisted of using an override and only the minimal field changes required from the upstream derivation

Reply:
OK great